packages/flutter/lib/src/material/menu_anchor.dart

```diff
@@ -1176,11 +1184,29 @@ class _MenuItemButtonState extends State<MenuItemButton> {
     }
   }
 
-  void _handleHover(bool hovering) {
-    widget.onHover?.call(hovering);
-    if (hovering && widget.requestFocusOnHover) {
-      assert(_debugMenuInfo('Requesting focus for $_focusNode from hover'));
-      _focusNode.requestFocus();
+  void _handlePointerExit(PointerExitEvent event) {
+    if (_isHovered) {
+      widget.onHover?.call(false);
+      _isHovered = false;
+    }
+  }
+
+  // TextButton.onHover and MouseRegion.onHover can't be used without triggering
+  // focus on scroll.
+  void _handlePointerHover(PointerHoverEvent event) {
+    if (!_isHovered) {
+      _isHovered = true;
+      widget.onHover?.call(true);
+      if (widget.requestFocusOnHover) {
+        assert(_debugMenuInfo('Requesting focus for $_focusNode from hover'));
+        _focusNode.requestFocus();
+
+        // Without invalidating the focus policy, switching to directional focus
+        // may not originate at this node.
+        FocusTraversalGroup.of(context).invalidateScopeData(
```

Comment:
This isn't all bad.  The default implementation is a no-op, and if directional traversal hasn't happened, then it's empty anyhow. The main use of the scope data is to keep the previous nodes that have been visited since the last invalidation so that when you traverse one direction and reverse, that it takes the same path on the way back as it did on the way there (to avoid hysteresis). Since menus kind of only have one path here, it's probably not going to matter if you nuke that information.